File: src/main/java/net/lomeli/trophyslots/core/command/CommandRemoveAll.java
package net.lomeli.trophyslots.core.command;

import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.ChatComponentText;
import net.minecraft.util.ChatComponentTranslation;
import net.minecraft.util.StatCollector;

import net.lomeli.trophyslots.TrophySlots;
import net.lomeli.trophyslots.core.SlotUtil;
import net.lomeli.trophyslots.core.network.MessageSlotsClient;

public class CommandRemoveAll extends CommandBase {
    @Override
    public String getCommandName() {
        return "remove-all";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 2;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "command.trophyslots.remove-all.usage";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if (args.length == 1 || args.length == 2) {
            EntityPlayerMP player = args.length == 2 ? getPlayer(sender, args[1]) : getCommandSenderAsPlayer(sender);
            if (player != null) {
                if (SlotUtil.getSlotsUnlocked(player) == 0)
                    sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.remove-all.error"), player.getDisplayName())));
                else {
                    SlotUtil.setSlotsUnlocked(player, 0);
                    TrophySlots.packetHandler.sendTo(new MessageSlotsClient(0), player);
                    player.addChatMessage(new ChatComponentTranslation("msg.trophyslots.lostAll"));
                    sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.remove-all.success"), player.getDisplayName())));
                }
            }
        } else
            sender.addChatMessage(new ChatComponentTranslation(getCommandUsage(sender)));
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/core/command/CommandRemoveAll.kt
package net.lomeli.trophyslots.core.command

import net.lomeli.trophyslots.capabilities.slots.SlotManager
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.server.MinecraftServer
import net.minecraft.util.text.TextComponentString
import net.minecraft.util.text.TextComponentTranslation
import net.minecraft.util.text.translation.I18n

class CommandRemoveAll : CommandBase() {
    override fun execute(server: MinecraftServer?, sender: ICommandSender?, args: Array<out String>?) {
        if (sender != null) {
            if (args != null && (args.size == 1 || args.size == 2)) {
                val player: EntityPlayerMP
                if (args.size == 2)
                    player = CommandBase.getPlayer(server, sender, args[1])
                else
                    player = CommandBase.getCommandSenderAsPlayer(sender)
                if (player != null) {
                    val slotInfo = SlotManager.getPlayerSlotInfo(player)!!
                    if (slotInfo.getSlotsUnlocked() <= 0)
                        sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.remove_all.error").format(player.displayName.unformattedText)))
                    else {
                        slotInfo.setSlots(0)
                        SlotManager.updateClient(player, slotInfo)
                        player.sendMessage(TextComponentTranslation("msg.trophyslots.lost_all"))
                        sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.remove_all.success").format(player.displayName.unformattedText)))
                    }
                }
            } else
                sender.sendMessage(TextComponentTranslation(getUsage(sender)))
        }
    }

    override fun getName(): String? = "remove-all"

    override fun getUsage(sender: ICommandSender?): String? = "command.trophyslots.remove_all.usage"

    override fun getRequiredPermissionLevel(): Int = 2
}